crates/jstz_proto/src/executor/fa_deposit.rs

```diff
@@ -0,0 +1,406 @@
+use crate::{
+    context::{account::Amount, ticket_table::TicketTable},
+    executor::smart_function,
+    operation::{external::FaDeposit, RunFunction},
+    receipt::Receipt,
+    Result,
+};
+use http::{header::CONTENT_TYPE, HeaderMap, Method, Uri};
+use jstz_api::http::body::HttpBody;
+use jstz_core::{host::HostRuntime, kv::Transaction};
+use jstz_crypto::{hash::Blake2b, public_key_hash::PublicKeyHash};
+use serde::{Deserialize, Serialize};
+use tezos_smart_rollup::prelude::debug_msg;
+
+const FA_DEPOSIT_GAS_LIMIT: usize = usize::MAX;
+
+// TODO: https://linear.app/tezos/issue/JSTZ-36/use-cryptos-from-tezos-crypto
+// Properly represent the null address
+const NULL_ADDRESS: &str = "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx";
+const DEPOSIT_URI: &str = "/-/deposit";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FaDepositReceiptContent {
+    pub receiver: PublicKeyHash,
+    pub ticket_balance: Amount,
+    pub run_function: Option<crate::receipt::RunFunction>,
+}
+
+fn deposit_to_receiver(
+    rt: &mut impl HostRuntime,
+    tx: &mut Transaction,
+    receiver: &PublicKeyHash,
+    ticket_hash: &Blake2b,
+    amount: Amount,
+) -> Result<FaDepositReceiptContent> {
+    let final_balance = TicketTable::add(rt, tx, receiver, ticket_hash, amount)?;
+    Ok(FaDepositReceiptContent {
+        receiver: receiver.clone(),
+        ticket_balance: final_balance,
+        run_function: None,
+    })
+}
+
+fn new_run_function(http_body: HttpBody, proxy_contract: &PublicKeyHash) -> RunFunction {
+    let mut headers = HeaderMap::new();
+    headers.insert(
+        CONTENT_TYPE,
+        "application/json; charset=utf-8".parse().unwrap(),
+    );
+    RunFunction {
+        uri: Uri::builder()
+            .scheme("tezos")
+            .authority(proxy_contract.to_string())
+            .path_and_query(DEPOSIT_URI)
+            .build()
+            .unwrap(),
+        method: Method::POST,
+        headers,
+        body: http_body,
+        gas_limit: FA_DEPOSIT_GAS_LIMIT,
+    }
+}
```

Comment:
It would be better to propagate the errors using a `Result` -- we should avoid `unwrap` or `expect` in the kernel code (as this may lead to panics)

Reply:
Returned result instead.